### ISSUE TITLE
PyType cannot handle type annotations of TypeAliasType (treated as callable)

### ISSUE THREAD

**Issue by wyattscarpenter** [google/pytype#1775]
[TypeAliasType](https://docs.python.org/3/library/typing.html#typing.TypeAliasType) is the type of what you get when you use the `type` keyword, eg:
```python3
type Point = tuple[float, float]
```

It is valid (or at least mypy and pyright allow you to) to pass the resultant type alias (`Point` in the example above) to functions, in which case it is supposed to have a type of TypeAliasType, However, this produces an error in PyType:

```python3
from typing import TypeAliasType

def f (t: TypeAliasType) -> None:
  pass
 ```
 
 here is the error:
```python3 
tat.py:3:1: error: in <module>: Invalid type annotation '<instance of Callable>'  [invalid-annotation]
  Not a type

def f (t: TypeAliasType) -> None:
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  pass
~~~~~~
```

Pyright and mypy don't error on this code, as I've implied.

Possibly related issues:

https://github.com/google/pytype/issues/1475 (I guess Python 3.12 support is not truly complete until you can do this?)

https://github.com/google/pytype/issues/1295 (possibly similar decay-to-callable issue?)

**Comment by frigus02** [comment]
Yes, Python 3.12 is only supported with Python 3.11 syntax right now.

**Comment by h-joo** [comment]
Hi, thank you for your report. Google is shifting its effort to a different approach for type checking apart from pytype, and we're not planning to put in any effort in the near future in pytype, so we'll close this issue. Please see the announcement [here](https://github.com/google/pytype/blob/main/README.md).
